### PR TITLE
Rename channel -> destination in StreamParser

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- All the references to `channel` in the StreamParser now changed to `destination`

This resolves spring-cloud/spring-cloud-dataflow#379

### CODE REVIEW

File: spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/ModuleDefinitionBuilder.java [hunk 1]
```diff
@@ -98,15 +98,15 @@ else if (argument.getName().equalsIgnoreCase("outputType")) {
 			}
 			builders.add(builder);
 		}
-		SourceChannelNode sourceChannel = streamNode.getSourceChannelNode();
+		SourceDestinationNode sourceChannel = streamNode.getSourceDestinationNode();
```
Comment (markfisher) [inline spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/ModuleDefinitionBuilder.java]: should change the variable names too... basically a case-insensitive search for "channel" in this file should find nothing

Comment (ilayaperumalg) [inline spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/ModuleDefinitionBuilder.java]: ah, missed those in my search. fixing it.

**Comment by ilayaperumalg** [comment]
Updated the PR with few additional renaming changes.

**Comment by markfisher** [comment]
rebased and merged as https://github.com/spring-cloud/spring-cloud-dataflow/commit/b336633d03726e4744a8580fb6079dd74f17056d

**Pull request closed without merging**